sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -349,14 +375,17 @@ public InputStream newInputStream(Path path, OpenOption... options) throws IOExc
      */
     @Override
     public OutputStream newOutputStream(Path path, OpenOption... options) throws IOException {
+        return newOutputStreamInternal(path, new HashSet<>(Arrays.asList(options)));
+    }
+
+    OutputStream newOutputStreamInternal(Path path, Set<? extends OpenOption> optionsSet,
+        FileAttribute<?>... fileAttributes) throws IOException {
         // If options are empty, add Create, Write, TruncateExisting as defaults per nio docs.
-        if (options == null || options.length == 0) {
-            options = new OpenOption[] {
-                StandardOpenOption.CREATE,
+        if (optionsSet == null || optionsSet.size() == 0) {
```

Comment:
Should we make the default option set a constant instead of instantiating a new set every time we have no options passed